src/Orchard.Web/Modules/Orchard.Projections/Services/ProjectionManager.cs

```diff
@@ -204,11 +209,11 @@ public IEnumerable<IHqlQuery> GetContentQueries(QueryPartRecord queryRecord, IEn
 
             var version = queryRecord.VersionScope.ToVersionOptions();
 
-            // pre-executing all groups
-            foreach (var group in queryRecord.FilterGroups) {
+            // Iterate over each filter group, but ignore empty ones, because they'd cause all content items to be returned.
+            foreach (var group in queryRecord.FilterGroups.Where(group => group.Filters.Count > 0)) {
```

Comment:
This is the only functional change, the rest is just code styling and comment rephrasing.

Reply:
I am suggesting adding a limit filter instead (100, 500). And add a notification.

Reply:
I didn't find an acceptable way to apply such a limit due to how the query is constructed (there is no explicit limit filter). It is possible to intervene at the point, where Slice is called (and apply a limit there), but that would mess up paging. I added a notification instead, when a Query or a ProjectionPart is saved, IMO that's good enough.